### ISSUE TITLE
Move RST files to subdirectories to make cloned jupyter notebooks easier to use

### ISSUE THREAD

**Issue by jlperla** [QuantEcon/lecture-source-jl#523]
Look at the datascience, for example.

The big thing this will change is the `:doc:` will all be out of wack.

**Comment by jlperla** [comment]
The basic directory structure should exactly match the sections of the TOC in https://github.com/jlperla/ECON307_2019

So I would have the following subfolder names would match the `_index.rst` names.
- `getting_started_julia`
- `more_julia`
- `tools_and_techniques`
- `dynamic_programming`
- `multi_agent_models`
- `time_series_models`
- `dynamic_programming_squared`

What to do for this:
- [ ] Create those subfolders in the RST source
- [ ] Move the appropriate RST for into each section, including the index file!
- [ ] Rename the index files for each section to just be `index.rst`.
For example, the current `rst_files/index_more_julia.rst` would become `rst_files/more_julia/index.rst`
- [ ] I am not sure what it at the top level in the folder, but it may just be
- The main `index.rst`, which points to the other indices
- `zreferences.rst`
- The `index_undergrad.rst` and `index_postgrad.rst`
- [ ] Go through all of the index files to make they are correct.  In particular, the top level index files will need to point at subfolders.  i.e. stuff like the current 
```
.. toctree::
    :maxdepth: 2
    :titlesonly:
    
    index_getting_started_julia
```
would become something like
```
.. toctree::
    :maxdepth: 2
    :titlesonly:
    
    getting_started_julia/index
```
- [ ] Go through all of the RST files which refer to other documents, and fix the links to deal with the reorganization.  This will take time, but you will need to globally search for the `:doc:`.  For example, I believe that the current reference inside of getting started
```
:doc:`version control <version_control>`
```
becomes something like
```
:doc:`version control <../more_julia/version_control>`
```
But using a `:doc:` to point at files in the same directory are not changed
- [ ] After all of that stuff is done, and you **carefully make sure that sphinx generates no errors!** go through every file and link you can image to ensure it works in the HTML and jupyter

**Comment by mmcky** [comment]
hey @jlperla I will look at this early next week to see easy this will be. This will be no problem from `sphinx` perspective -- it will be the supporting infrastructure for running notebooks that may need to change. Currently it expects a collection of `ipynb` notebooks as a flat structure using ``glob`` so that would need to be updated to include sub-directories (or a crawl of the directories).

**Comment by jlperla** [comment]
For what it is worth, this is the structure of the datascience lectures

**Comment by mmcky** [comment]
yeah - I just modified the `glob` to include a single layer of subdirectories for quantecon.build.datascience

**Comment by mmcky** [comment]
hey @arnavs would you have time to setup a branch and add the relevant lectures to a subdirectory? Then I can set about looking at the execution script to adjust it for the new directory structure.

**Comment by arnavs** [comment]
@mmcky Ready for you, in the `directory-refactor` branch.

**Comment by mmcky** [comment]
thanks @arnavs. I wonder where the `jl` went. It is correct in the `conf.py` with option: `jupyter_target_html_urlpath = "https://lectures.quantecon.org/jl/"`

**Comment by mmcky** [comment]
I can only take a quick look at this sometime this afternoon -- but I'll likely get some time in the week 4th Feb to fix this.

**Comment by mmcky** [comment]
@arnavs when compiling with the `html` target I am getting

```
Note that we assume you’ve already completed the [getting started](https://lectures.quantecon.org/jl/../getting_started_julia/getting_started) and [interacting with Julia](https://lectures.quantecon.org/jl/../getting_started_julia/julia_environment) lectures

**Comment by mmcky** [comment]
@arnavs I have submitted a fix for this on the extension. 

https://github.com/QuantEcon/sphinxcontrib-jupyter/pull/172

I did this very quickly. Once you have verified the `docker` build with the `build` server output -- would you mind comparing the branch of the sphinxextension above to see if this resolves the issues with paths (and more importantly doesn't introduce errors)

**Comment by mmcky** [comment]
Just a final comment on the above -- you can use the `diff` script to check the pre-executed stats of the generated notebooks. That is what I would recommend doing.

**Comment by arnavs** [comment]
- [ ] Verify this resolves path issue. 
- [ ] Once @PooyaFa is done with #527, can run testing against this branch of `sphinxcontrib-jupyter`.

**Comment by arnavs** [comment]
Closing in favor of the issue in Jupinx and the build infrastructure issue.
